File: lab_2/generator.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <ctime>

using namespace std;

vector<int> generateBinaryNumbers(int size) {
    vector<int> numbers(size);
    srand(time(0));
    for (int i = 0; i < size; ++i) {
        numbers[i] = rand() % 2;
    }
    return numbers;
}

int main() {
    int size = 128;
    vector<int> binaryNumbers = generateBinaryNumbers(size);
    for (int num : binaryNumbers) {
        cout << num;
    }
    cout << endl;
    return 0;
}

File: lab_2/generator.java
import java.util.Random;

public class Main {

    public static int[] generateBinaryNumbers(int size) {
        int[] numbers = new int[size];
        Random random = new Random();
        for (int i = 0; i < size; ++i) {
            numbers[i] = random.nextInt(2);
        }
        return numbers;
    }

    public static void main(String[] args) {
        int size = 128;
        int[] binaryNumbers = generateBinaryNumbers(size);
        for (int num : binaryNumbers) {
            System.out.print(num);
        }
    }
}
